### PR TITLE
[IMM32] Rewrite ImmGetDefaultIMEWnd

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Rewrite `ImmGetDefaultIMEWnd` function.
- Add `g_dwImm32Flags` global variable.
- Add `Imm32QueryWindow` helper function.
- Implement `CtfImmIsCiceroEnabled` function.
- Modify `imm32.spec`.

### CODE REVIEW

File: dll/win32/imm32/imm.c [hunk 1]
```diff
@@ -2096,14 +2106,21 @@ void WINAPI __wine_unregister_window(HWND hwnd)
...
+
+    return (HWND)Imm32QueryWindow(hWnd, QUERY_WINDOW_DEFAULT_IME);
+}
+
+/***********************************************************************
+ *		CtfImmIsCiceroEnabled (IMM32.@)
+ */
+BOOL WINAPI CtfImmIsCiceroEnabled(VOID)
+{
+    return (!!g_dwImm32Flags && !!(g_dwImm32Flags & IMM32_FLAG_CICERO_ENABLED));
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply dll/win32/imm32/imm.c:2123]: See https://reactos.org/wiki/User:Alvinhochun/Localization/IME#Text_Services_Frammework_.28TSF.29

[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/imm32/imm.c:2123]: Cicero is the codename of Text Services Frammework (TSF).
TSF is the advanced framework from old-style IME/IMM.

[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/imm32/imm.c:2123]: Side note: Both 'ime' and 'cicero' have negative meaning in Japanese language.

[the remark this answers is not in the document]
Comment (HBelusca) [reply dll/win32/imm32/imm.c:2123]: Cicero is the name of a Roman philosopher.

**Pull request merged**